src/MediaType.php

```diff
@@ -0,0 +1,390 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\MediaType;
+
+use Libero\MediaType\Exception\InvalidMediaType;
+use LogicException;
+use OutOfBoundsException;
+use function addcslashes;
+use function is_callable;
+use function mb_strlen;
+use function mb_strtolower;
+use function mb_substr;
+use function preg_match;
+
+final class MediaType
+{
+    private const HTTP_WHITESPACE = [' ', "\t", "\n", "\r"];
+
+    private $type;
+    private $subType;
+    private $parameters;
+
+    /**
+     * @param array<string, string> $parameters
+     */
+    private function __construct(string $type, string $subType, array $parameters)
+    {
+        $this->type = $type;
+        $this->subType = $subType;
+        $this->parameters = $parameters;
+    }
+
+    public static function fromString(string $input) : MediaType
+    {
+        // 1. Remove any leading and trailing HTTP whitespace from input.
```

Comment:
where do the comments come from? Is it forked from somewhere?

Reply:
Updated the opening comment, and about to update the readme.